torchinductor/dependencies.py

```diff
@@ -44,9 +46,9 @@ def maybe_swap_sizes(self) -> "MemoryDep":
         ):
             c = canonicalization_prefix()
             size = (self.size[1], self.size[0])
-            s0 = sympy.Symbol(c + "0", is_integer=True)
```

Comment:
Does `is_integer` slow it down?

Reply:
No, but:
```
>>> sympy.Symbol("x") == sympy.Symbol("x")
True
>>> sympy.Symbol("x") == sympy.Symbol("x", is_integer=True)
False
```

Reply:
🤔 Doesn't this indicate that we should just be setting `is_integer` to be True or False everywhere?

I'm worried if not setting `is_integer` prohibits certain simplifications.


For ex:

(Also, shouldn't we be setting `integer=True` and not `is_integer=True`? cc: @asmeurer )

```
>>> a = sympy.Symbol('a',positive=True)
>>> a * 2 > 1
2*a > 1
>>> a = sympy.Symbol('a', is_integer=True, positive=True)
>>> a * 2 > 1
2*a > 1
>>> a = sympy.Symbol('a', integer=True, positive=True)
>>> a * 2 > 1
True
```

Reply:
Yeah, I think we need to set it uniformly everywhere.

Currently, we make no usage of sympy inequalities, so I don't think this setting actually matters.